kernel/source/utils.c

```diff
@@ -22,26 +22,28 @@ Copyright (C) 2008, 2009	Hector Martin "marcan" <marcan@marcansoft.com>
 bool IsWiiMode = true;
 
 //no idea why this has less issues when forced as arm...
-__attribute__((target("arm")))
-void udelay(u32 delay)
+__attribute__((target("arm"))) void udelay(u32 delay)
 {
 	u32 ticks = ConvertDelayToTicks(delay);
-	if(ticks < 2)
+	if (ticks < 2)
 		ticks = 2;
-	
+
 	u32 then = read32(HW_TIMER) + ticks;
 
 	//There are 2 possibilities in waiting here
 	//either the HW_Timer value is smaller than our future time (regular wait)
 	//or our future overflowed and we have to wait on the timer to overflow back to 0
-	if(read32(HW_TIMER) < then)
+	if (read32(HW_TIMER) < then)
 	{
-		while(read32(HW_TIMER) < then);
+		while (read32(HW_TIMER) < then)
+			;
 		return;
 	}
 	else
 	{
-		while(read32(HW_TIMER) > 0);
-		while(read32(HW_TIMER) < then);
+		while (read32(HW_TIMER) > 0)
+			;
```

Comment:
also not sure if this is a setting?

Reply:
Setting `AllowShortLoopsOnASingleLine: true` fixed this.